### PR TITLE
Setup: Remove embeddings setup step UI 

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/53809

This merges the embedding flow and local repositories selection step into one; we limit repositories to up to 10 and automatically include all selected repos for embedding uploads.

## Test plan 
- Check that you're able to go through all setup steps and have embeddings working

### CODE REVIEW

**Comment by fkling** [comment]
Does that mean we are not listing repos grouped by folder anymore?

File: client/web/src/enterprise/app/setup/steps/AppLocalRepositoriesSetupStep.tsx [hunk 1]
```diff
@@ -84,3 +98,36 @@ export const AddLocalRepositoriesSetupPage: FC<StepComponentProps> = ({ classNam
...
+interface PathsPickerActionsProps {
+    disabled: boolean
+    className?: string
+    onPathsChange: (paths: string[]) => void
+}
+
+/**
+ * Local repositories path picker buttons, both do the same job,
+ * but we have two buttons to improve user understanding what options
+ * they have in the file picker.
```
Comment (fkling) [inline client/web/src/enterprise/app/setup/steps/AppLocalRepositoriesSetupStep.tsx:111]: There is only one button now.

**Comment by fkling** [comment]
Won't that change break user's existing setup (they might have a folder of repos configured already)? Given that it's possible to have that (plus the new external service config allows that too), I'd say we keep the UI for rendering folders of repos (unless there is a good reason not to do that).

**Comment by taylorsperry** [comment]
> Won't that change break user's existing setup (they might have a folder of repos configured already)? Given that it's possible to have that (plus the new external service config allows that too), I'd say we keep the UI for rendering folders of repos (unless there is a good reason not to do that).

The reason for removing the folder option is that we thought it might be tricky given our 10 repo limitation in this step. (We're capping at 10 repos because that's what we feel comfortable with given our $20/user embeddings rate limit). I wouldn't think that this change would break an existing user's setup, since only new users will walk through the setup flow, but I defer to @vovakulikov.

**Comment by vovakulikov** [comment]
@fkling, you're right; this may break things, to be more specific. 
- If you already have folders 
- And folders at the moment don't allow to remove repositories from folder one by one, but since we don't render folders in this PR delete button next to the repository item won't work 

I haven't thought about new API integration. Maybe with this new API, it won't be a problem, but if we can't merge it this week, I suggest that, for this case if we click on the delete button for the repo that previously was selected as part of the folder we will remove this repo and all other repositories that were in the folder and we have as a flat list at the moment, cc @taylorsperry 

Another suggestion would be to keep the limitation of ten repositories but don't change anything about folder repos representations, you still can select multiple repositories, but if you select more than 10 you will see in the UI that you can't completely this step with more than ten repos.

cc @taylorsperry @fkling

**Comment by taylorsperry** [comment]
cc @danielmarquespt for input on @vovakulikov 's proposed solutions?

**Comment by danielmarquespt** [comment]
> Won't that change break user's existing setup 

Sorry I read the comments but I don't understand how it impacts existing users.

Why would a existing user go through this flow again? This should be for the new comers only, right?

As far as I understand, we won't auto-update existing app users, so they only go through this if they download a new Cody.app version. In that case its fare that we start from a blank slate.

**Comment by taylorsperry** [comment]
> As far as I understand, we won't auto-update existing app users

We'll begin auto-updating again (starting with 6.16.2023 release, we just didn't want to autoupdate for users who might have been using code search and would be surprised to see "cody only"), but I'm still not sure I understand what break this would introduce since a user only goes through the setup experience once.

**Comment by taylorsperry** [comment]
My vote is for whatever is easiest/safest to ship for now so that we can be sure the combined step makes it in to @meglynst's research study (kicking off in ~4.5 hours)

**Comment by vovakulikov** [comment]
@fkling and I just had a pair session and this is something that we got

- App data is preserved across installation this mean that user that picked repositories on one version of the app will see these repositories again on the new version (only hard reinstall I think remove persist data from the machine)
- Local reposiotires selection widget is also used on the setting page and not having there folders breaks repositories deletion (see my comment above about it)

So even users see setup flow once I don't think this is safe to remove folders from the picker. Based on our meeting with Felix our plan is 

- Double down on Felix's PR about new API (that simplify work with repositories on the backend) 
- Based on Felix's PR I will try to add embeddings uploading (as I did in this PR) 

From product side this means that we still would have folders UI, and user still can pick more than 10 repositories in the picker but if it's the case we will prevent this in the setup flow by disabling next and file picker button and it's presented in the design.

**Comment by chwarwick** [comment]
> * Double down on Felix's PR about new API (that simplify work with repositories on the backend)

We need to be cautious about this as it's quite close to the final cut to risk in switching over how we deal with repositories and the integration with the extensions is fragile around repo name. 

@fkling  What are the specific benefits that we pickup from the new service? 

@taylorsperry  Is giving users from previous versions a smooth upgrade experience a priority or are we stating that what we release on the 28th is the new starting point and that if prior to that you had the app installed it should now be uninstalled?

**Comment by taylorsperry** [comment]
Ideally, anyone who's on 6.16.23 or later (the release that introduced Cody-only) will be able to upgrade without having to uninstall.

**Comment by fkling** [comment]
@chwarwick The immediate benefit would be having git clone the repos locally, in which case it would use hardlinks if possible (at least that's what I gather from the docs). That would both reduce the amount of disk space needed and speed.

If we don't care about existing configurations (or find a way to migrate existing ones) we can also disable the `servegit` service, which means one less port to open.

The repo name issue is something I can look into, but I understand the concerns.

**Comment by chwarwick** [comment]
> @chwarwick The immediate benefit would be having git clone the repos locally, in which case it would use hardlinks

Ah so does this means instead of making a 2nd copy over in  `{CACHE_DIR}/repos` that is will just use the path it already sits in? That would be a very nice improvement and possibly help with some of the issues with trying to schedule embeddings jobs.   [This](https://github.com/sourcegraph/sourcegraph/pull/53610) has the naming logic if you want to try to pull it to the service.

**Comment by fkling** [comment]
@chwarwick No, we'll still clone the repo (I haven't touched the logic that clones repos) but since it uses a `file:` URL, git has the ability to do some optimizations (whether that actually happens or not idk).

**Pull request merged**